### PR TITLE
[SPARK-41734][CONNECT] Add a parent message for Catalog

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to add a parent Protobuf message for Catalog (see https://github.com/apache/spark/pull/39214#discussion_r1057439608).

### Why are the changes needed?

For better readability, and we could maintain this more easily in the future. We should redesign this part.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unittests should cover them

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zhengruifeng and @grundprinzip FYI

**Review by zhengruifeng — APPROVED** [review]

File: connector/connect/common/src/main/protobuf/spark/connect/catalog.proto [hunk 1]
```diff
@@ -24,6 +24,41 @@ import "spark/connect/types.proto";
 option java_multiple_files = true;
 option java_package = "org.apache.spark.connect.proto";
 
+// Catalog messages are all meant to be internal.
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/catalog.proto:27]:
```suggestion
// Catalog messages are marked as unstable..
```

File: connector/connect/common/src/main/protobuf/spark/connect/relations.proto [hunk 2]
```diff
@@ -70,35 +70,7 @@ message Relation {
     StatDescribe describe = 102;
 
     // Catalog API (internal-only)
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/relations.proto:72]:
```suggestion
    // Catalog API (experimental / unstable)
```

**Review by grundprinzip — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**